osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -33,6 +33,14 @@ public class ChannelManager : PollingComponent, IChannelPostTarget
         private readonly BindableList<Channel> availableChannels = new BindableList<Channel>();
         private readonly BindableList<Channel> joinedChannels = new BindableList<Channel>();
 
+        // Keeps a list of closed channels. More recently closed channels appear at higher indeces
+        private readonly BindableList<Channel> closedChannels = new BindableList<Channel>();
```

Comment:
This should be only storing the name or ID, not the full channel object. This is going to be keeping messages, users, etc. in memory forever.